ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/KDCKerberosOperationHandler.java

```diff
@@ -225,8 +248,14 @@ protected Keytab createKeytab(String principal, String password, Integer keyNumb
     return super.executeCommand(command, _envp, interactiveHandler);
   }
 
-  String getAdminServerHost() {
-    return adminServerHost;
+  /**
+   * Returns the KDC administration server host value (with or without the port)
+   *
+   * @param includePort <code>true</code> to include the port (if available); <code>false</code> to exclude the port
+   * @return the KDC administration server host value (with or without the port)
+   */
+  String getAdminServerHost(boolean includePort) {
+    return (includePort) ? adminServerHostAndPort : adminServerHost;
```

Comment:
It feels like this is a bit error prone to have a boolean control whether you get server:port or just server.  I actually don't clearly see why `invokeKAdmin` wants server:port while `getKinitCommand` does not. Would it be clearer to break this out into 2 properties?

Reply:
For most cases, the host and port are needed to declare where the server is. However in a special case, we need to not have the port since we are using the data to build a service principal name.   

I guess that I can break this into two methods. I originally did this and then changed it. I figured someone would see the getAdminServerHost method and not notice the getAdminServerHostAndPort method or not know to get the port using some getAdminServerPort method.  Also since most of the time the hostname and port values were to be used together, I would have that value cached and ready to go rather then recalculating the string every time.

Reply:
Hmmm - OK, we can keep it as one. It was just confusing to me when you'd need one vs the other.